checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub

```diff
@@ -0,0 +1,215 @@
+    boolean retainAll(Collection<?> c);
+}
+
+public interface Deque<E> extends Queue<E> {
+    boolean removeFirstOccurrence(@Nullable Object o);
+    boolean removeLastOccurrence(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean contains(@Nullable Object o);
+}
+
+public interface List<E> extends Collection<E> {
+    boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean containsAll(Collection<?> c);
+    boolean removeAll(Collection<?> c);
+    boolean retainAll(Collection<?> c);
+    int indexOf(@Nullable Object o);
+    int lastIndexOf(@Nullable Object o);
+}
+
+public interface Map<K, V> {
+    boolean containsKey(@Nullable Object key);
+    boolean containsValue(@Nullable Object value);
+    @Nullable V get(@Nullable Object key);
+    @Nullable V remove(@Nullable Object key);
+    default V getOrDefault(@Nullable Object key, V defaultValue);
+    default boolean remove(@Nullable Object key, @Nullable Object value);
+    }
+
+public interface Set<E> extends Collection<E> {
+    @Pure boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    @Pure boolean containsAll(Collection<?> c);
+    boolean retainAll(Collection<?> c);
+    boolean removeAll(Collection<?> c);
+}
+
+class SynchronousQueue {
+    @Pure public boolean containsAll(Collection<?> c);
+    public boolean removeAll(Collection<?> c);
+    public boolean retainAll(Collection<?> c);
+}
+
+public class TreeMap<K,V>
+    extends AbstractMap<K,V>
+    implements NavigableMap<K,V>, Cloneable, java.io.Serializable
+{
+    public boolean containsKey(@Nullable Object key);
+    public boolean containsValue(@Nullable Object value);
+    public @Nullable V get(@Nullable Object key);
+    public @Nullable V remove(@Nullable Object key);
+}
+
+public class TreeSet<E> extends AbstractSet<E>
+    implements NavigableSet<E>, Cloneable, java.io.Serializable
+{
+    public boolean contains(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+}
+
+class Vector {
+    public synchronized boolean containsAll(Collection<?> c);
+    public synchronized boolean removeAll(Collection<?> c);
+    public synchronized boolean retainAll(Collection<?> c);
+}
+
+package java.util.concurrent;
+
+public class ConcurrentLinkedDeque<E>
+    extends AbstractCollection<E>
+    implements Deque<E>, java.io.Serializable {
+    public boolean removeFirstOccurrence(@Nullable Object o);
+    public boolean removeLastOccurrence(@Nullable Object o);
+}
+
+public interface ConcurrentMap<K,V> extends Map<K,V> {
+    @Nullable V putIfAbsent(@Nullable K key, @Nullable V value);
+    boolean remove(@Nullable Object key, @Nullable Object value);
+}
+
+public class ConcurrentSkipListSet<E>
+    extends AbstractSet<E>
+    implements NavigableSet<E>, Cloneable, java.io.Serializable {
+    public boolean contains(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+}
+
+public class CopyOnWriteArrayList<E>
```

Comment:
If you don't want to permit `null` in `CopyOnWriteArrayList.containsAll`+`removeAll`+`retainAll` in the annotated JDK, then you could do it here.

I haven't grepped through to find all classes that override those methods to see which others might benefit from similar changes; I just happened to notice this one.)

Reply:
You're generally postponing work on `removeAll` and `retainAll`, but you could consider handling `containsAll` now if you wanted.

I think that's the only thought I have left!

Reply:
I added `removeAll` and `retainAll` to the stub file (maximally permissively), but I'm deferring work on them in the annotated JDK.  I made `containsAll` more permissive, either in the annotated JDK or in the stub file, depending on the specification.